src/page/product-list.page.ts

```diff
@@ -1,14 +1,29 @@
-import { $, ElementFinder, browser, ExpectedConditions } from 'protractor';
+import { $, ElementFinder, browser, ExpectedConditions, ElementArrayFinder, $$ } from 'protractor';
 
 export class ProductListPage {
   private productList: ElementFinder;
-
+  private products: ElementArrayFinder;
   constructor () {
     this.productList = $('a.button.ajax_add_to_cart_button.btn.btn-default');
+    this.products = $$('product-container');
```

Comment:
can you explain to me what the `$$` does? for what instruction is this a shortener?

Reply:
I understood is a shortcut notation when we want to get an array of items, instead of use element.all(by.css()). So, in this case,  due to we want to get all the products, is necesary to use this notation that will allow us select an array instead of a particular element.

Reply:
> You are missing step 3. That requires changes in the test class:
> 
> > Elimine el método que antes obtenía el primer elemento y cambielo por un método llamado selectProduct que reciba el nombre del producto y le da clic en la imagen

In this step, we have to click the image and the button 'add to cart' ?